src/array/cuda/sddmm.cu

```diff
@@ -91,6 +91,43 @@ void SDDMMCsr(const std::string& op,
   });
 }
 
+/*!
+ * \brief CUDA implementation of g-SDDMM on heterograph using 
+    Csr format.
+ */
+template <int XPU, typename IdType, int bits>
+void SDDMMCsrHetero(const std::string& op,
+              const BcastOff& bcast,
+              const std::vector<CSRMatrix>& vec_csr,
+              std::vector<NDArray> vec_lhs,
+              std::vector<NDArray> vec_rhs,
+              std::vector<NDArray> vec_out,
+              int lhs_target,
+              int rhs_target,
+              const std::vector<dgl_type_t> lhs_eid,
+              const std::vector<dgl_type_t> rhs_eid) {
```

Comment:
We may want to pass all of these `std::vector`s as `const std::vector<>&`, since we don't want to copy them.